### PR TITLE
Organize playwright test tapes into folders

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Related to #797 by @sarayourfriend
Not strictly related, but adding this change would greatly simplify navigating the tapes while working on #797.
Marked as high because further work on testing should be based on this change.

## Description

This PR moves all the tapes used in the Playwright tests into folders based on the request type: `detail`, `search`, `related` and `thumb`, and on the media type inside those folders. This means we can hide the folder and don't have to scroll as much in the IDE file list.
It also splits the visual regression tests into two folders: pages and components.

## Testing Instructions

Passing CI should be enough.
The main change is in `test/proxy.js` file.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by sarayourfriend** [comment]
@obulat why are the snapshots changing in this one? :thinking:

**Review by sarayourfriend — APPROVED** [review]
LGTM! Much more organized.

Shall we add a README describing the folders and when to put things in the components vs the page tests? Additionally, when to choose to test a component on an actual page instead of writing a story for it and writing the e2e tests for it there.

File: test/proxy.js [hunk 1]
```diff
@@ -26,10 +26,12 @@ const port = 49153
...
-  search: /\/(?<mediaType>images|audio)\/*\?(?<query>[\w&=]+)/,
-  thumb: /\/(?<mediaType>images|audio)\/(?<uuid>[\w-]{32,})\/thumb/,
-  related: /\/(?<mediaType>images|audio)\/(?<uuid>[\w-]{32,})\/related/,
-  detail: /\/(?<mediaType>images|audio)\/(?<uuid>[\w-]{32,})\//,
+  search: /\/(?<mediaType>images|audio|video|model-3d)\/*\?(?<query>[\w&=]+)/,
+  thumb:
+    /\/(?<mediaType>images|audio|video|model-3d)\/(?<uuid>[\w-]{32,})\/thumb/,
+  related:
+    /\/(?<mediaType>images|audio|video|model-3d)\/(?<uuid>[\w-]{32,})\/related/,
+  detail: /\/(?<mediaType>images|audio|video|model-3d)\/(?<uuid>[\w-]{32,})\//,
```
[the remark this answers is not in the document]
Comment (obulat) [reply test/proxy.js:34]: Yes, and I tried doing so, but I wasn't very successful :) I cannot import constants from a TypeScript file into `proxy.js` 😭

[the remark this answers is not in the document]
Comment (sarayourfriend) [reply test/proxy.js:34]: Ah, of course! Maybe we should just use ts-node after all? It's pretty frustrating to not be able to use some parts of our code. Alternatively we can convert the media constants to a typed JS file.

[the remark this answers is not in the document]
Comment (obulat) [reply test/proxy.js:34]: I remember I didn't have a very nice experience with `ts-node` a year ago, so I'm not very fond of this solution. Converting the media constants to a typed JS file sounds a simpler way out ...

[the remark this answers is not in the document]
Comment (sarayourfriend) [reply test/proxy.js:34]: > Converting the media constants to a typed JS file sounds a simpler way out

Me too! I think that's the consensus I've gathered from everyone about `ts-node`, though I've seen some projects use it with success! It just incurs some weird maintenance overhead sometimes. Maybe something to explore in the future but JSDoc TS works just fine for 99% of non-complicated types :slightly_smiling_face:

[the remark this answers is not in the document]
Comment (obulat) [reply test/proxy.js:34]: I've opened an issue for that: #1518

**Pull request merged**